src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -519,7 +527,6 @@ private void ParseStatusLine(Span<byte> line, HttpResponseMessage response)
             response.Version =
                 (majorVersion == '1' && minorVersion == '1') ? HttpVersionInternal.Version11 :
                 (majorVersion == '1' && minorVersion == '0') ? HttpVersionInternal.Version10 :
-                (majorVersion == '2' && minorVersion == '0') ? HttpVersionInternal.Version20 :
```

Comment:
No strong opinion, but should the response.Version accurately reflect what was sent back by the server, even if it was erroneous? I'm wondering if in addition to removing this line, we should change the final else to just construct an HttpVersion with the supplied values.

Also, I think we were using HttpVersion call because of 2.0. Now that it's being removed, can we switch to using HttpVersion instead?

Reply:
Yeah, I agree.  That said, it's not a huge issue, as in practice it should never actually happen... but still seems like a good thing to do.

> Also, I think we were using HttpVersion call because of 2.0. Now that it's being removed, can we switch to using HttpVersion instead?

Do you mean, we should switch from HttpVersionInternal to HttpVersion?  Yeah, we can do that.  I filed an issue on this somewhere...

Reply:
> Do you mean, we should switch from HttpVersionInternal to HttpVersion

Sorry, yes, writing with non dominant hand on phone :)

Reply:
In reading the RFC (https://tools.ietf.org/html/rfc7230#section-2.6), I think we should be a bit more restrictive here.  In particular, we should fail if major = 0 (which is undefined) or major >= 2 (which implies incompatibility with 1.x).  

So that means that, for example, 1.2 is still valid and we should construct a valid response Version for this.

Reply:
It looks like @caesar1995 is making some related test changes here.